File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.java
class Solution {
    public int minAddToMakeValid(String s) {
        int open = 0;
        int close = 0;
        
        for (char ch : s.toCharArray()) {  
            if (ch == '(') {
                open++;
            } else if (ch == ')' && open > 0) {
                open--;
            } else if (ch == ')') {
                close++;
            }
        }
        return open + close;
    }
}

File: 957-minimum-add-to-make-parentheses-valid/minimum-add-to-make-parentheses-valid.py
class Solution:
    def minAddToMakeValid(self, s: str) -> int:
        stack = []
        for i in range(len(s)):
            stack.append(s[i])
            if len(stack) >= 2:
                if stack[-1] == ')' and stack[-2] == '(':
                    stack.pop()
                    stack.pop()
        return len(stack)            
